libraries/standard/mqtt/cbmc/stubs/network_interface_stubs.c

```diff
@@ -67,18 +67,15 @@ int32_t NetworkInterfaceSendStub( NetworkContext_t * pNetworkContext,
 
     /* If the maximum tries are reached, then return a timeout. In the MQTT library
      * this stub is wrapped in a loop that will does not end until the bytesOrError
-     * returned is zero or less. This means we could loop possibly INT32_MAX
+     * returned is negative. This means we could loop possibly INT32_MAX
      * iterations. Looping for INT32_MAX times adds no value to the proof.
      * What matters is that the MQTT library can handle all the possible values
      * that could be returned. */
-    if( tries >= MAX_NETWORK_SEND_TRIES )
+    tries++;
+    if( tries >= ( MAX_NETWORK_SEND_TRIES - 1 ) )
```

Comment:
It would be a bit cleaner to write
```
if( tries < MAX_NETWORK_SEND_TRIES ) {increment} else {zero}
```

Reply:
If we do that then we call the function MAX_NETWORK_SEND_TRIES (3) instead of MAX_NETWORK_SEND_TRIES - 1 (2) times still.

Reply:
Unless I start tries at 1 instead of 0.